tutorials/securing-your-app-with-spring-security/04__form-less/content.adoc

```diff
@@ -0,0 +1,141 @@
+@Route(value = LoginView.ROUTE)
+@PageTitle("Login")
+public class LoginView extends VerticalLayout {
+    public static final String ROUTE = "login";
+
+    private LoginOverlay login = new LoginOverlay(); // <1>
+
+    @Autowired
+    public LoginView(AuthenticationManager authenticationManager, // <2>
+                     CustomRequestCache requestCache) {
+        // configures login dialog and adds it to the main view
+        login.setOpened(true);
+        login.setTitle("Spring Secured Vaadin");
+        login.setDescription("Login Overlay Example");
+
+        add(login);
+
+        login.addLoginListener(e -> { // <3>
+            try {
+                // try to authenticate with given credentials, should always return !null or throw an {@link AuthenticationException}
+                final Authentication authentication = authenticationManager
+                    .authenticate(new UsernamePasswordAuthenticationToken(e.getUsername(), e.getPassword())); // <4>
+
+                // if authentication was successful we will update the security context and redirect to the page requested first
+                SecurityContextHolder.getContext().setAuthentication(authentication); // <5>
+                login.close(); // <6>
+                UI.getCurrent().navigate(requestCache.resolveRedirectUrl()); // <7>
+
+            } catch (AuthenticationException ex) { // <8>
+		// show default error message
+		// Note: You should not expose any detailed information here like "username is known but password is wrong"
+		// as it weakens security.
+                login.setError(true);
+            }
+        });
+    }
+}
+----
+<1> Let's use the awesome login dialog component Vaadin provides.
+<2> Inject needed beans. The `authenticationManager` to kindly ask for login validation and the `requestCache` to get the redirect URL.
+<3> The component allows registering a login listener that gives access to the provided username and password.
+<4> Starts the authentication process by creating an authentication request object and let the manager do the rest. If successful we get a fully configured authentication object.
+<5> We have to register the authentication object in the security context manually to make Spring Security happy and aware of it.
+<6> If the authentication was successful we must not forget to close the dialog. Otherwise you will not see much of your views.
+<7> Resolve the redirect URL and route to the location.
+<8> In cases the authentication failed, we will inform the user about it via the dialog. It is always a good practice to give as less information as possible.
+
+That's it. Now run `mvn spring-boot:run` and open localhost:8080. You will be redirected to the login view, should be able to provide the credentials and redirected back to the root. You can also try URLs like localhost:8080/path-not-exists. As we did not implement any sophisticated redirect checkers you will be faced with Vaadin's default 404 page.
```

Comment:
"and redirected back to the root" -> confusing since the sentence starts with "you will be redirected to the login view". If you really mean redirected a second time then emphasize sequence as : 
"and redirected back" -> "and subsequently be redirected back"